docs/api/framework-conventions/entry.server.tsx.md

```diff
@@ -0,0 +1,141 @@
+---
+title: entry.server.tsx
+order: 5
+---
+
+# entry.server.tsx
+
+[MODES: framework]
+
+## Summary
+
+<docs-info>
+This file is optional
+</docs-info>
+
+This file is the server-side entry point that controls how your React Router application generates HTTP responses on the server.
+
+This module should render the markup for the current page using a [`<ServerRouter>`][serverrouter] element with the `context` and `url` for the current request. This markup will (optionally) be re-hydrated once JavaScript loads in the browser using the [client entry module][client-entry].
+
+## Generating `entry.server.tsx`
+
+By default, React Router will handle generating the HTTP Response for you. You can reveal the default entry server file with the following:
+
+```shellscript nonumber
+npx react-router reveal
+```
+
+## Exports
+
+### `default`
+
+The `default` export of this module is a function that lets you create the response, including HTTP status, headers, and HTML, giving you full control over the way the markup is generated and sent to the client.
+
+```tsx filename=app/entry.server.tsx
+import { renderToString } from "react-dom/server";
+import { ServerRouter } from "react-router";
+
+export default function handleRequest(
+  request: Request,
+  responseStatusCode: number,
+  responseHeaders: Headers,
+  routerContext: EntryContext
+) {
+  const markup = renderToString(
```

Comment:
I thought that with single fetch (which is the only option in RR framework mode) we can’t use “renderToString”? 🤔

https://remix.run/docs/en/main/guides/single-fetch

Can’t deep link - 5th point in “Enabling single fetch”